mlflow/utils/cli_args.py

```diff
@@ -45,12 +48,86 @@
 NO_CONDA = click.option(
     "--no-conda",
     is_flag=True,
-    help="If specified, will assume that MLmodel/MLproject is running within "
+    help="[Deprecated] If specified, will assume that MLmodel/MLproject is running within "
     "a Conda environment with the necessary dependencies for "
     "the current project instead of attempting to create a new "
     "conda environment.",
 )
 
+
+def _env_manager_callback(_, __, value):
+    """
+    Validates the value of `--env-manager` and converts it to the corresponding member of
+    `EnvManager`.
+    """
+    if value is None:
+        return value
+
+    allowed_values = [e.value for e in EnvManager]
+    if value not in allowed_values:
+        raise click.BadParameter(f"Expected one of {allowed_values} but got '{value}'")
+
+    return EnvManager[value.upper()]
+
+
+ENV_MANAGER = click.option(
+    "--env-manager",
+    default=None,
+    required=False,
+    type=click.UNPROCESSED,
+    callback=_env_manager_callback,
+    help="If specified, create an environment for MLmodel/MLproject using the specified "
+    "environment management tool. Valid values are ['local', 'conda', 'virtualenv']. "
+    "If unspecified, default to 'conda'.",
+)
+
+
+def _get_env_manager(no_conda, env_manager):
+    result = env_manager
+    # Both `--no-conda` and `--env-manager` are specified
+    if no_conda and env_manager is not None:
+        raise Exception(
+            "`--no-conda` (deprecated) and `--env-manager` cannot be specified at the same time."
+        )
+    # Only `--no-conda` is specified
+    elif no_conda:
+        warnings.warn(
+            (
+                "`--no-conda` is deprecated and will be removed in a future MLflow release. "
+                "Please use `--env-manager=local` instead."
+            ),
+            FutureWarning,
+            stacklevel=2,
+        )
+        result = EnvManager.LOCAL
+    # Neither `--no-conda` nor `--env-manager` is specified. In this case, conda should be used
+    # to preserve the existing behavior.
+    elif env_manager is None:
+        result = EnvManager.CONDA
+
+    if result is EnvManager.CONDA:
+        warnings.warn(
+            (
+                "Restoring the model environment using conda is discouraged now. Please use "
+                "virtualenv instead by specifying `--env-manager=virtualenv`."
+            ),
+            UserWarning,
+            stacklevel=2,
+        )
```

Comment:
I think your other PR removed this, but we shouldn't say that conda use is discouraged until 2.x.

Reply:
ok, removed :)